pandas/_testing.py

```diff
@@ -299,39 +309,35 @@ def write_to_compressed(compression, path, data, dest="test"):
     ------
     ValueError : An invalid compression value was passed in.
     """
-    if compression == "zip":
-        compress_method = zipfile.ZipFile
-    elif compression == "gzip":
-        # pandas\_testing.py:288: error: Incompatible types in assignment
-        # (expression has type "Type[GzipFile]", variable has type
-        # "Type[ZipFile]")
-        compress_method = gzip.GzipFile  # type: ignore[assignment]
-    elif compression == "bz2":
-        # pandas\_testing.py:290: error: Incompatible types in assignment
-        # (expression has type "Type[BZ2File]", variable has type
-        # "Type[ZipFile]")
-        compress_method = bz2.BZ2File  # type: ignore[assignment]
-    elif compression == "xz":
-        compress_method = get_lzma_file(lzma)
-    else:
-        raise ValueError(f"Unrecognized compression type: {compression}")
+    compress_method = _select_compress_method(compression)
```

Comment:
i think you could leave this inline and just type compress_method no?

Reply:
I guess so.
I just thought that some small function extraction will be helpful here.
But if you suggest against the extraction, I can definitely do that way as well.

Reply:
I would prefer for this to stay inline as well

Reply:
I inlined the function and also slid the conditionals. Can revert if this is not what you expect, but in my opinion, it simplifies the understanding that ``zip`` has a different signature as compared to other compression methods.